extensions/who-is-visiting-this-content/app.R

```diff
@@ -0,0 +1,151 @@
+library(shiny)
+library(bslib)
+library(shinyjs)
+library(connectapi)
+library(dplyr)
+library(glue)
+library(lubridate)
+library(tidyr)
+
+shinyOptions(
+  cache = cachem::cache_disk("./app_cache/cache/", max_age = 60 * 60 * 8)
+)
+
+source("get_usage.R")
+
+ui <- page_fillable(
+  useShinyjs(),
+  theme = bs_theme(version = 5),
+
+  card(
+    card_header("Who is Visiting This Content?"),
+    layout_sidebar(
+      sidebar = sidebar(
+        title = "No Filters Yet",
+        open = FALSE,
+
+        actionButton("clear_cache", "Clear Cache", icon = icon("refresh"))
+      ),
+
+      textInput(
+        "content_guid",
+        "Content GUID"
+      ),
+
+      h4(
+        id = "guid_input_msg",
+        "Please enter a content GUID"
+      ),
+
+      textOutput("summary_message"),
+
+      tabsetPanel(
+        id = "content_visit_tables",
+        tabPanel(
+          "List of Visits",
+          tableOutput("all_visits")
+        ),
+        tabPanel(
+          "Aggregated Visits",
+          tableOutput("aggregated_visits")
+        )
+      )
+    )
+  )
+)
+
+server <- function(input, output, session) {
+  # Cache invalidation button ----
+  cache <- cachem::cache_disk("./app_cache/cache/")
+  observeEvent(input$clear_cache, {
+    print("Cache cleared!")
+    cache$reset()  # Clears all cached data
+    session$reload()  # Reload the app to ensure fresh data
+  })
+
+  observe({
+    if (nchar(input$content_guid) == 0) {
+      show("guid_input_msg")
+      hide("content_visit_tables")
+    } else {
+      hide("guid_input_msg")
+      show("content_visit_tables")
+    }
+  })
+
+  # Loading and processing data ----
+  client <- connect()
+
+  # Default dates. "This week" is best "common sense" best represented by six
+  # days ago thru the end of today. Without these, content takes too long to
+  # display on some servers.
+  date_range <- reactive({
+    list(
+      from_date = today() - ddays(6),
+      to_date = today()
+    )
+  })
```

Comment:
This is a great default + kick the can of "which is the best period? Should I do forever? etc" for when we get to the filtering edition later 💯